articles/client-platforms/angular2/04-linking-accounts.md

```diff
@@ -0,0 +1,206 @@
+
+Let's make it work. As you need to do a second login to get the secondary account JWT, we will use another instance of `AuthLock`. This is to differentiate the login from the linking login.
+
+As all instances listen to the `authenticated` event (we are in redirect mode, so we don’t have a reliable way to determine which instance did the login) we need a way to know if the login came from a login or from a linking login. We can use [params auth option](https://github.com/auth0/lock/tree/v10.0.0-rc.1#authentication-options), setting a `state` property to `"linking"`.
+
+```typescript
+// Lock instance to lauch a login to obtain the secondary JWT
+lockLink = new Auth0Lock('${account.clientId}', '${account.namespace}', {
+    auth: {params: {state: "linking"}},
+    allowedConnections: ['Username-Password-Authentication', 'facebook', 'google-oauth2'],
+    languageDictionary: { // allows to override dictionary entries
+      title: "Link with:"
+    }
+  });
+```
+
+Then, when setting the `authenticated` callbacks we can know which login is, checking the `authResult.state` attribute. 
+
+```typescript
+...
+// Add callback for lock `authenticated` event
+this.lock.on("authenticated", (authResult) => {
+  // Every lock instance listen to the same event, so we have to check if
+  // it's not the linking login here.
+  if(authResult.state != "linking"){
+    localStorage.setItem('id_token', authResult.idToken);
+    this.fetchProfile(authResult.idToken);
+  }
+});
+
+// Add callback for lockLink `authenticated` event
+this.lockLink.on("authenticated", (authResult) => {
+  // Every lock instance listen to the same event, so we have to check if
+  // it's the linking login here.
+  if(authResult.state == "linking"){
+    // If it's the linking login, then do the link through the API.
+    this.doLinkAccounts(authResult.idToken);
+  }
+});
+...
+```
+
+Now that we already have the second login handled let's see how to actually do the linking.
+To call the api, `angular-jwt` provides a `AuthHttp` class which automatically attach the user's JWT as an `Authorization` header when making requests.
+
+First you need to add the AUTH_PROVIDERS from angular-gwt
+
+${snippet(meta.snippets.auth-providers-setup)}
+
+Then you can import AuthHttp in your component and make the authenticated request:
+
+
+```typescript
+public doLinkAccounts(accountToLinkJWT) {
+  var headers: any = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
+  };
+
+  var data: any = JSON.stringify({
+    link_with: accountToLinkJWT
+  });
+
+  this.authHttp
+    .post('https://' + '${account.namespace}' + '/api/v2/users/' + this.userProfile.user_id + '/identities', data, {headers: headers})
```

Comment:
It could be, but I like as it cause it's clearer that you have some variable value there, what do you think?